lib/active_hash/base.rb

```diff
@@ -218,13 +218,20 @@ def field(field_name, options = {})
         field_names << field_name
 
         add_default_value(field_name, options[:default]) if options[:default]
-        define_getter_method(field_name, options[:default])
+        define_getter_method(field_name, options)
         define_setter_method(field_name)
         define_interrogator_method(field_name)
         define_custom_find_method(field_name)
         define_custom_find_all_method(field_name)
       end
 
+      def already_defined?(method_name)
```

Comment:
@djberg96 I really like this and the changes to `instance_methods.include?` below
There is another PR out there that reports that the code sometimes overwrites methods.

Do you think it is worth it to get these changes in?